src/client/_clientwindow.py

```diff
@@ -870,12 +874,25 @@ def saveWindow(self):
         util.settings.setValue("geometry", self.saveGeometry())
         util.settings.endGroup()
 
+    def show_autojoin_settings_dialog(self):
+        autojoin_channels_list = Settings.get('chat/auto_join_channels')
+        text_of_autojoin_settings_dialog = """
+        Autojoin channels list : """ + str(autojoin_channels_list) + """
```

Comment:
There's still no need to put the current list separately here. We can pass the default textbox contents through the 'text' parameter, as shown [here](http://doc.qt.io/qt-5/qinputdialog.html#getText). Some ';'.join(list) should be enough.

Reply:
Ugh, I think I did what you asked but not sure what you meant...?

Reply:
I think it's better to set the textbox contents to current settings rather than describe the settings in the box label. Lose the 'Autojoin channels list:' line, and instead add the joined channel list as the fifth argument to getText (or equivalently, a 'text' keyword argument).

Reply:
Ughh..wut? Sorry I don't get what you mean....

For the fifth argument of getText, I did this : 
channels_input_of_user, ok = QInputDialog.getText(self, 'Set autojoin channels',text_of_autojoin_settings_dialog, QLineEdit.Normal, 'Autojoin channels list : ' + ';'.join(autojoin_channels_list))
But then the line where the user can write the channels has inside this text 'autojoin channels list' which is not good. (added the join channel list as fifth argument to this gettext).

For the box label : 
For me the box label is what is shown to the user, which is the 3rd argument of gettext. And I set the content of the textbox (this popup) to the text inside this variable text_of_autojoin_settings_dialog.
 And what do you mean current settings? what is that?

Reply:
That first effect you're describing (putting autojoin_channels_list in the box where the user writes stuff) is exactly what I'm aiming for. It feels natural to me that the contents of these box are at the same time the value of the autojoin_channels_list setting. You get this association that what is in the box is the current setting, and what you change is what the new setting will be. It also makes it easier for the user to edit an existing setting - he doesn't have to clumsily copy over the text from the box label to the edited line.

Reply:
Ooh, ok ! Got it.